data/json/items/gun/303.json

```diff
@@ -0,0 +1,57 @@
+[
+  {
+    "id": "smle_mk3",
+    "copy-from": "rifle_manual",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "Enfield battle rifle" },
+    "description": "An iconic weapon of both the First and Second World Wars, this bolt-action rifle has seen deployment in the hands of Commonwealth troops across Europe and beyond.  Outfitted with a 17-inch bayonet, firing the .303 calibre cartridge, and making use of 10-round Enfield magazines, the firearm’s reputation precedes it.",
+    "//": "Total unloaded weight of an Mk III is 3.96 kg, and a No. 4 is 4.11 kg.  total length of an Mk III is 1132mm, and a No. 4 is 1129mm.  Rounding to 4 kg of weight and 1130mm of length to meet in the middle.",
+    "weight": "4000 g",
+    "volume": "3000 ml",
+    "longest_side": "1130 mm",
+    "price": 57375,
+    "price_postapoc": 3000,
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "smle_mk3",
+        "name": { "str": "SMLE No. 1 Mk III" },
+        "description": "A venerable English darling and icon of the First World War, the Short Magazine Lee-Enfield Mk III was the standard issue battle rifle of the British commonwealth from 1907.  Chambered for .303 ammunition and assigned with a sword-style bayonet, the rifle makes use of ten-round detachable magazines.  With its full length wooden handguard, brass buttplate, and smooth action, the Lee-Enfield cuts a recognizable profile in either the trenches of the Great War or the zombie-strewn wastelands.",
+        "weight": 1
+      },
+      {
+        "id": "number4_mki",
+        "name": { "str": "Lee-Enfield No. 4 Mk I" },
+        "description": "A Second World War progression of the original Lee-Enfield battle rifle adopted by the British Commonwealth; the No. 4 began its decades-long service life in 1939.  Externally similar to its forefather, the rifle possesses a reinforced receiver, adjusted nose cap, and heavier barrel.  Still chambered for the same .303 caliber cartridge, the No. 4 represents one of the most populous firearm imports in U.S. history."
+      }
+    ],
+    "to_hit": -1,
+    "bashing": 12,
+    "material": [ "steel", "wood" ],
+    "color": "brown",
+    "ammo": [ "303" ],
+    "ranged_damage": { "damage_type": "bullet", "amount": 9 },
+    "dispersion": 100,
+    "durability": 8,
+    "blackpowder_tolerance": 60,
+    "barrel_volume": "250 ml",
+    "default_mods": [ "sword_bayonet" ],
```

Comment:
The issue of the sword bayonet is honestly an interesting one, cause I don’t think that you can define a probability for it’s spawn within the nested item group. Looking at stuff like the mod attachments for the M38 DMR, it looks like that they’d always spawn anyway, which means that defining the bennet in a nested item group as apposed to a default mod would have no functional difference, save for the amount of JSON present. That being said, theoretically I could define a nested item group and spawn for  a lee-Enfield without the attachment, and another, rarer spawn for an Enfield with a bayonet, but then this would honestly be more lines of JSON than needed, and would bring into question all the other weapons that also spawn with a bayonet by default, like the 1896 trench gun.
That being said, splitting the No. 4 and Mk III into two items, the former with the spike bayonet depicted here, is a decent idea. The only change I’d suggest of doing for the spike is giving it piercing damage rather than cutting.

Reply:
the "STAB" flag gives it piercing damage

Reply:
I see. I'll get around to adding it when I've got more time to work on this Pr, but thanks for the info and the framework for what the attachment should look like.